File: Movie/Movie.Web/Modules/ServerTypes/Club/EmployeeForm.ts
import { StringEditor, DateEditor, LookupEditor, EnumEditor, DecimalEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { ClubDialog } from "../../Club/Club/ClubDialog";
import { Gender } from "../Modules/MovieDB.Person.Gender";

export interface EmployeeForm {
    FirstName: StringEditor;
    LastName: StringEditor;
    BirthDate: DateEditor;
    Address: StringEditor;
    ClubId: LookupEditor;
    Gender: EnumEditor;
    Salary: DecimalEditor;
}

export class EmployeeForm extends PrefixedContext {
    static readonly formKey = 'Club.Employee';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmployeeForm.init)  {
            EmployeeForm.init = true;

            var w0 = StringEditor;
            var w1 = DateEditor;
            var w2 = LookupEditor;
            var w3 = EnumEditor;
            var w4 = DecimalEditor;

            initFormType(EmployeeForm, [
                'FirstName', w0,
                'LastName', w0,
                'BirthDate', w1,
                'Address', w0,
                'ClubId', w2,
                'Gender', w3,
                'Salary', w4
            ]);
        }
    }
}

queueMicrotask(() => [ClubDialog, Gender]); // referenced types

File: Movie/Movie.Web/Modules/Club/Employee/EmployeeForm.cs
using Movie.Modules.MovieDB.Person;
using Serenity.ComponentModel;
using System;

namespace Movie.Club.Forms;

[FormScript("Club.Employee")]
[BasedOnRow(typeof(EmployeeRow), CheckNames = true)]
public class EmployeeForm
{
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public DateTime BirthDate { get; set; }
    public string Address { get; set; }

    public int ClubId { get; set; }

    public Gender Gender { get; set; }
    public decimal Salary { get; set; }
}